packages/starlight/schema.ts

```diff
@@ -117,7 +117,9 @@ type BaseSchemaWithoutEffects =
 type BaseSchema = BaseSchemaWithoutEffects | z.ZodEffects<BaseSchemaWithoutEffects>;
 
 /** Type that extends Starlight’s default schema with an optional, user-defined schema. */
-type ExtendedSchema<T extends BaseSchema> = T extends BaseSchema
+type ExtendedSchema<T extends BaseSchema | never = never> = [T] extends [never]
```

Comment:
This [pattern](https://www.typescriptlang.org/docs/handbook/2/conditional-types.html#distributive-conditional-types) is used to disable distributivity as when conditional types act on a generic type, they become distributive when given a union type which can be the case here.